unified-runtime/source/adapters/opencl/usm.cpp

```diff
@@ -294,9 +294,10 @@ UR_APIEXPORT ur_result_t UR_APICALL urEnqueueUSMFill(
   // not a power of 2, we need to do on the host side and copy it into the
   // target allocation.
   clHostMemAllocINTEL_fn HostMemAlloc = nullptr;
-  UR_RETURN_ON_FAILURE(cl_ext::getExtFuncFromContext<clHostMemAllocINTEL_fn>(
-      CLContext, ur::cl::getAdapter()->fnCache.clHostMemAllocINTELCache,
-      cl_ext::HostMemAllocName, &HostMemAlloc));
+  bool HostAllocSupported =
```

Comment:
we need to check the result against UR_RESULT_SUCCESS, casting it directly to bool has the opposite effect (UR_RESULT_SUCCESS == 0 == false)